README.md

```diff
@@ -72,9 +72,52 @@ You can also use the [onnx-dev docker image](https://hub.docker.com/r/onnx/onnx-
 
 ## Build ONNX from Source
 Before building from source uninstall any existing versions of onnx `pip uninstall onnx`.  
-If you are building ONNX from source, it is recommended that you also build Protobuf locally as a static library. Specifically on Windows, the version distributed with conda-forge is a DLL, but ONNX expects it to be a static library. Building protobuf locally also let's you control the verison of protobuf. The tested and recommended version is 3.16.0.
 
-Note for Windows : The instructions in this README assume you are using Visual Studio. It is recommended that you run all the commands from a shell started from "Developer Command Prompt for VS 2019" and keep the build system generator for cmake (e.g., cmake -G "Visual Studio 16 2019") consistent while building protobuf as well as ONNX.
+Generally spreaking, you need to install [protobuf C/C++ libraires and tools](https://github.com/protocolbuffers/protobuf) before proceeding forward. Then depending on how you installed protobuf, you need to set environment variable CMAKE_ARGS to "-DONNX_USE_PROTOBUF_SHARED_LIBS=ON" or "-DONNX_USE_PROTOBUF_SHARED_LIBS=OFF".  For example, you may need to run the following command:
+
+Linux:
+```bash
+export CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=ON"
+```
+Windows:
+```bat
+set CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=ON"
+```
+
+The ON/OFF depends on whether your protobuf was built as a shared library or static library, or both. If you have both, then we recommend you setting it to OFF for Windows and ON for Linux.
+
```

Comment:
why ON for Linux?

Reply:
what about Mac?

Reply:
Because most shared libraries are built without "-fPIC". And this sentence is for people have both static and shared libraries. In that case, pretty much the static one won't work.

Reply:
I don't know much about Mac.

Reply:
Maybe I should change the wording to:
```
What if you have both:
Window: we recommend  you using the static one.
Linux: if your static library was built without "-fPIC"(the most common case), then it can't be used by onnx. If you don't know how it was built, let assume it doesn't have the flag.  Otherwise we recommend you using the static one.
```

Reply:
Or just remove the line:
~If you have both, then we recommend you setting it to OFF for Windows and ON for Linux.~

Reply:
if the shared lib is built without -fPIC meaning it is NOT position independent then this is a problem right... and in this case using static lib should be the right way

Reply:
Shared lib won't be.

Reply:
Meaning? Are you saying we can assume shared lib will be build with -fPIC becasue this is the default these days? then yes I agree. 
Now I understand the instructions better... earlier I did not quite understand what was the intention.

May be we can change the instructions to: 

What if you have both:
Window: we recommend  you using the static one.
Linux: When building from source we recommend using static library following the instructions in this doc. If using prebuilt libs, we recommend using shared lib.